internal/v2/tls_config_store/tls_config_store.go

```diff
@@ -186,20 +184,54 @@ func ClientConfig(tokenManager tokenmanager.AccessTokenManager, localIdentities
 
 		clientAuth := getTLSClientAuthType(tlsConfig)
 
+		var cipherSuites []uint16
+		if maxVersion <= tls.VersionTLS12 {
```

Comment:
nit: please omit the TLS version check, just to future proof against e.g. configurable ciphersuites in TLS 1.3.

Reply:
Good point! Done!